vm/src/ser_de.rs

```diff
@@ -0,0 +1,199 @@
+use std::fmt;
```

Comment:
It would be neat if we could use this same code for `marshall.rs`, but then use bincode instead of json!